sites/org/pages/blog/page/[page].mjs

```diff
@@ -72,42 +73,59 @@ const Preview = ({ post, t }) => (
  * when path and locale come from static props (as here)
  * or set them manually.
  */
-const BlogIndexPage = ({ page, posts }) => {
+const BlogIndexPage = ({ posts, page, current, total }) => {
   const { t } = useTranslation()
+
   return (
     <PageWrapper {...page}>
       <div className="grid grid-cols-1 gap-4 lg:grid-cols-2 xl:grid-cols-3 max-w-7xl lg:pr-4 xl:pr-6">
         {posts.map((post) => (
-          <Preview post={post} t={t} key={post.slug} />
+          <Preview post={post} t={t} key={post.slug.current} />
         ))}
       </div>
+      <Pagination {...{ current, total }} />
     </PageWrapper>
   )
 }
 
 export default BlogIndexPage
 
-export async function getStaticProps({ locale }) {
-  const posts = await sanityLoader({ language: locale, type: 'blog' }).catch((err) =>
-    console.log(err)
-  )
+export async function getStaticProps({ locale, params }) {
+  const allPosts = await sanityLoader({
+    language: locale,
+    type: 'blog',
+    order: 'date desc',
+    filters: '{_id, date, slug, title, author, image}',
+  })
+  const pageNum = parseInt(params.page)
 
   return {
     props: {
-      posts: posts.map((post) => ({
-        slug: post.slug.current,
-        title: post.title,
-        date: post.date,
-        // FIXME get the authors separately
-        author: post.author,
-        image: sanityImage(post.image) + '?fit=clip&w=400',
-      })),
+      posts: allPosts.slice(numPerPage * (pageNum - 1), numPerPage * pageNum),
+      current: pageNum,
+      total: allPosts.length,
       ...(await serverSideTranslations(locale, namespaces)),
       page: {
         locale,
         // title: 'Freesewing Blog',
-        path: ['blog'],
+        path: ['blog', 'page', params.page],
       },
     },
   }
 }
+
+const locales = ['', 'de', 'es', 'fr', 'nl']
```

Comment:
Please load the locales from `site/site.config.mjs` and filter out `en` rather than maintain a list of locales here too.